### PR TITLE
Added eslint config

### PR DESCRIPTION
**Pull request by Revadike**
A proposed config for formatting the code

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        "no-new-object": "error",
+        "no-new": "warn",
+        "no-octal-escape": "error",
+        "no-octal": "error",
+        "no-param-reassign": "warn",
+        "no-proto": "error",
+        "no-redeclare": "error",
+        "no-return-assign": "error",
+        "no-return-await": "error",
+        "no-script-url": "warn",
```
Comment (antigravities) [inline .eslintrc.json]: If we're making `no-eval` an `error`, this should also be an `error`.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        "no-confusing-arrow": ["warn", { "allowParens": true }],
+        "no-constructor-return": "error",
+        "no-dupe-else-if": "error",
+        "no-else-return": "warn",
+        "no-empty-function": "error",
+        "no-eq-null": "warn",
+        "no-eval": "error",
+        "no-floating-decimal": "warn",
+        "no-implicit-coercion": "warn",
+        "no-implied-eval": "warn",
```
Comment (antigravities) [inline .eslintrc.json]: If we're making `no-eval` an `error`, this should also be an `error`.

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        "prefer-const": "warn",
+        "prefer-exponentiation-operator": "warn",
+        "prefer-regex-literals": "warn",
+        "prefer-rest-params": "error",
+        "prefer-template": "warn",
+        "quote-props": ["error", "always"],
+        "quotes": ["error", "double", { "avoidEscape": true }],
+        "require-await": "warn",
+        "rest-spread-spacing": "error",
+        "semi-spacing": "error",
```
Comment (antigravities) [inline .eslintrc.json]: the behavior defined here may be confusing to someone reading the file, maybe write the definition as `["error", "last"]` instead

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        "default-param-last": "error",
+        "dot-location": ["error", "property"],
+        "dot-notation": "error",
+        "eol-last": "error",
+        "eqeqeq": ["error", "smart"],
+        "func-call-spacing": "error",
+        "generator-star-spacing": "error",
+        "grouped-accessor-pairs": ["error", "getBeforeSet"],
+        "implicit-arrow-linebreak": "error",
+        "indent": ["error", 4, { "SwitchCase": 1 }],
```
Comment (antigravities) [inline .eslintrc.json]: don't trigger me. if this is an actual suggestion, 2 spaces, no tabs. it's enough to see what is going on without taking up too much space.

**Comment by Revadike** [comment]
oh wow, it got merged before I even got the change to go through all of the reviews and fix it

File: .eslintrc.json (hunk 4 above)
Comment (Revadike) [reply .eslintrc.json]: you must get triggered a lot, cuz 4 spaces is by far the most popular one

File: .eslintrc.json (hunk 3 above)
Comment (Revadike) [reply .eslintrc.json]: I don't see `"last"` as a valid argument for [this option](https://eslint.org/docs/rules/semi-spacing).

**Comment by bartervg** [comment]
move fast and merge stuff

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    "rules": {
+        "arrow-spacing": "error",
+        "block-scoped-var": "error",
+        "block-spacing": "error",
+        "brace-style": ["error", "1tbs", { "allowSingleLine": true }],
+        "camelcase": "warn",
+        "comma-spacing": ["warn", { "before": false, "after": true }],
+        "comma-style": "error",
+        "consistent-return": "error",
+        "consistent-this": ["error", "that"],
```
[the remark this answers is not in the document]
Comment (Revadike) [reply .eslintrc.json]: Well, that's easy. `this` can refer to different things in different contexts/scopes. By assigning it to a variable, you can refer to the correct context, consistently. And it just looks better if that variable is named the same everywhere.

File: .eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        "func-call-spacing": "error",
+        "generator-star-spacing": "error",
+        "grouped-accessor-pairs": ["error", "getBeforeSet"],
+        "implicit-arrow-linebreak": "error",
+        "indent": ["error", 4, { "SwitchCase": 1 }],
+        "key-spacing": "error",
+        "keyword-spacing": "error",
+        "lines-between-class-members": "error",
+        "max-len": ["error", 120],
+        "max-params": ["warn", 4],
```
[the remark this answers is not in the document]
Comment (Revadike) [reply .eslintrc.json]: I think having too many parameters hurts the reliability of the code. Instead, a better practice in that case would be to pass the information into an object.

**Pull request merged**